### PR TITLE
Fixes for symbolic arrays change

### PR DESCRIPTION
**Pull request by shashi**
Running MTK off of https://github.com/JuliaSymbolics/Symbolics.jl/pull/197

### CODE REVIEW

File: src/systems/diffeqs/odesystem.jl [hunk 1]
```diff
@@ -86,8 +86,8 @@ function ODESystem(
                    connection_type=nothing,
                   )
     iv′ = value(iv)
-    dvs′ = value.(dvs)
-    ps′ = value.(ps)
+    dvs′ = value.(collect(dvs))
+    ps′ = value.(collect(ps))
```
Comment (YingboMa) [inline src/systems/diffeqs/odesystem.jl:90]: Does `collect` work recursively?

Comment (shashi) [reply src/systems/diffeqs/odesystem.jl:90]: no

Comment (YingboMa) [reply src/systems/diffeqs/odesystem.jl:90]: Then how would `collect(dvs)` work for `[A[1:3] x]`?

Comment (shashi) [reply src/systems/diffeqs/odesystem.jl:90]: ideally `[A[1:3]; x]` should be a `vcat` node but we don't have that yet. I would imagine Julia generics will take care of making that into an array and collect will be a no-op.

Comment (YingboMa) [reply src/systems/diffeqs/odesystem.jl:90]: Collect on an array is a copy though. Maybe call `collect` only when `dvs` is a symbolic array?

File: src/parameters.jl [hunk 2]
```diff
@@ -21,6 +31,16 @@ Maps the variable to a state.
 tovar(s::Symbolic) = setmetadata(s, MTKParameterCtx, false)
 tovar(s::Num) = Num(tovar(value(s)))
 
+function recurse_and_apply(f, x)
```
Comment (YingboMa) [inline src/parameters.jl:34]: Looks like this function is never called.

File: src/parameters.jl [hunk 3]
```diff
@@ -10,7 +10,17 @@ isparameter(x) = false
...
 Maps the variable to a paramter.
 """
-toparam(s::Symbolic) = setmetadata(s, MTKParameterCtx, true)
+function toparam(s)
+    if s isa Symbolics.Arr
+        Symbolics.wrap(toparam(Symbolics.unwrap(s)))
+    elseif s isa AbstractArray
+        map(toparam, s)
+    elseif symtype(s) <: AbstractArray
+        Symbolics.recurse_and_apply(toparam, s)
```
Comment (YingboMa) [inline src/parameters.jl:19]: Is it moved to Symbolics?

Comment (shashi) [reply src/parameters.jl:19]: no we copied it here.

Comment (shashi) [reply src/parameters.jl:19]: but forgot to use the copy lol

Comment (YingboMa) [reply src/parameters.jl:19]: Maybe change it?

**Comment by YingboMa** [comment]
DataDrivenDiffEq needs a separate update. It tries to use implementation details to handle array variables.

**Pull request merged**